spyder/app/mainwindow.py

```diff
@@ -1983,9 +1996,21 @@ def reset_spyder(self):
             self.restart(reset=True)
 
     @Slot()
-    def restart(self, reset=False):
+    def restart_in_debug_mode(self):
+        """
+        Quit and restart Spyder in debug mode.
+        """
+        answer = QMessageBox.warning(self, _("Warning"),
+             _("Spyder will restart in debug mode: <br><br>"
+               "Do you want to continue?"),
+             QMessageBox.Yes | QMessageBox.No)
+        if answer == QMessageBox.Yes:
+            self.restart(debug=True)

```

Comment:
This can be moved to the Application plugin.

Reply:
Should the reset_spyder just above it also be moved to the Application plugin?

Reply:
Yeah, it makes sense too.